### PR TITLE
ENH/API: better defaults for .apply_function and co (closes  #3742)

### LINKED ISSUE #3742: API: default params in `raw.apply_hilbert` and `raw.apply_funtion`
These two methods would be more handsome if equipped with some defaults.
At least picks and n_jobs could default to something, maybe also dtype.
What do you think? @Eric89GXL @agramfort ...

### PR DESCRIPTION
**Pull request by dengemann**
cc @agramfort @Eric89GXL

### CODE REVIEW

File: mne/io/base.py [hunk 1]
```diff
@@ -809,13 +810,13 @@ def apply_function(self, fun, picks, dtype, n_jobs, *args, **kwargs):
             A function to be applied to the channels. The first argument of
             fun has to be a timeseries (numpy.ndarray). The function must
             return an numpy.ndarray with the same size as the input.
-        picks : array-like of int | None
+        picks : array-like of int (defaul: None)
             Indices of channels to apply the function to. If None, all
-            M-EEG channels are used.
-        dtype : numpy.dtype
+            M-EEG channels are used. If None, all data channels are used.
+        dtype : numpy.dtype (default: None)
```
Comment (agramfort) [inline mne/io/base.py]: it's not None by default

File: mne/io/base.py [hunk 2]
```diff
@@ -850,9 +856,10 @@ def apply_function(self, fun, picks, dtype, n_jobs, *args, **kwargs):
                                       for p in picks)
             for pp, p in enumerate(picks):
                 self._data[p, :] = data_picks_new[pp]
```
Comment (agramfort) [inline mne/io/base.py]: where do you check that the passed dtype is ok?

Comment (dengemann) [inline mne/io/base.py]: why should we check now? we did not check before.

**Comment by dengemann** [comment]
Fixed default type. @agramfort

File: mne/io/base.py (hunk 2 above)
Comment (agramfort) [inline mne/io/base.py]: if you have a default now we should check that the default value is ok no?

Comment (dengemann) [inline mne/io/base.py]: But the default is handled internally already, it tests for `None` and other types.

**Comment by agramfort** [comment]
ok fair enough. Please have a look why appveyor is not happy

**Comment by dengemann** [comment]
>  Please have a look why appveyor is not happy

Everyone is happy, appveyor too.

**Comment by agramfort** [comment]
ok cool. Please merge when Travis is green

File: mne/io/base.py [hunk 3]
```diff
@@ -809,13 +810,13 @@ def apply_function(self, fun, picks, dtype, n_jobs, *args, **kwargs):
             A function to be applied to the channels. The first argument of
             fun has to be a timeseries (numpy.ndarray). The function must
             return an numpy.ndarray with the same size as the input.
-        picks : array-like of int | None
+        picks : array-like of int (defaul: None)
             Indices of channels to apply the function to. If None, all
-            M-EEG channels are used.
-        dtype : numpy.dtype
+            M-EEG channels are used. If None, all data channels are used.
```
Comment (jaeilepp) [inline mne/io/base.py]: ``defaul`` -> ``default``. Also remove the sentence in the middle.

Comment (jaeilepp) [inline mne/io/base.py]: Never mind. I'll fix it later.

**Pull request merged**